pydantic/types.py

```diff
@@ -107,6 +121,60 @@ def validate(cls, value):
         return validate_email(value)[1]
 
 
+class UrlStr(str):
+    strip_whitespace = True
```

Comment:
Maybe I'm being dumb, but it looks like `stripe_whitespace = False` won't be respected.

Reply:
Why? I'm using validators `anystr_strip_whitespace` and `anystr_length_validator` and here a simple example that strip is working:

```
>>> from pydantic import BaseModel, UrlStr
>>> class Model(BaseModel):
...     v: UrlStr
...
>>>
>>> Model(v='http://example.com    ').v
'http://example.com'
>>>
```